DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -0,0 +1,342 @@
+  setMETitle(muEta_," mu_#eta","events / ");
+
+  histname = "mu_d0"; histtitle = "mu_d0";
+  bookME(ibooker,mud0_,histname,histtitle, d0_binning_.nbins,d0_binning_.xmin, d0_binning_.xmax);
+  setMETitle(mud0_," mu_d0","events /1cm ");
+
+  histname = "mu_z0"; histtitle = "mu_z0";
+  bookME(ibooker,muz0_,histname,histtitle, z0_binning_.nbins,z0_binning_.xmin, z0_binning_.xmax);
+  setMETitle(muz0_," mu_z0","events /1cm ");
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void BPHMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+//  edm::Handle<reco::PFMETCollection> metHandle;
+//  iEvent.getByToken( metToken_, metHandle );
+//  reco::PFMET pfmet = metHandle->front();
+//  if ( ! metSelection_( pfmet ) ) return;
+  
+//  float met = pfmet.pt();
+//  float phi = pfmet.phi();
+
+//  edm::Handle<reco::PFJetCollection> jetHandle;
+//  iEvent.getByToken( jetToken_, jetHandle );
+//  std::vector<reco::PFJet> jets;
+//  if ( int(jetHandle->size()) < njets_ ) return;
+//  for ( auto const & j : *jetHandle ) {
+//    if ( jetSelection_( j ) ) jets.push_back(j);
+//  }
+//  if ( int(jets.size()) < njets_ ) return;
+  
+//  edm::Handle<reco::GsfElectronCollection> eleHandle;
+//  iEvent.getByToken( eleToken_, eleHandle );
+//  std::vector<reco::GsfElectron> electrons;
+//  if ( int(eleHandle->size()) < nelectrons_ ) return;
+//  for ( auto const & e : *eleHandle ) {
+//    if ( eleSelection_( e ) ) electrons.push_back(e);
+//  }
+//  if ( int(electrons.size()) < nelectrons_ ) return;
+  
+  edm::Handle<reco::MuonCollection> muoHandle;
+  iEvent.getByToken( muoToken_, muoHandle );
+  if ( int(muoHandle->size()) < nmuons_ ) return;
+  std::vector<reco::Muon> muons;
+  for ( auto const & m : *muoHandle ) {
+    if ( muoSelection_( m ) ) muons.push_back(m);
+  }
+  if ( int(muons.size()) < nmuons_ ) return;
+  for (int i=0;i<muons.size()+1;i++) {
+    muPhi_.denominator->Fill(muons[i].phi());
+    muEta_.denominator->Fill(muons[i].eta());
+    muPt_.denominator ->Fill(muons[i].pt());
+    const Track * track = 0;
+    if (muons[i].isTrackerMuon()) track = & * muons[i].innerTrack();
+    else if (muon.isStandAloneMuon()) track = & * muons[i].outerTrack();
+    if (track) {
+      double d0 = track->dxy(beamSpot->position());
```

Comment:
are you sure you want to compute the impact parameter w.r.t. the beamspot ?
I thought you want w.r.t. the PV, but I might be wrong